templates/terraform/flatten_property_method.erb

```diff
@@ -30,7 +30,7 @@ func flatten<%= prefix -%><%= titlelize_property(property) -%>(v interface{}, d
   }
   <% end -%>
   transformed := make(map[string]interface{})
-  <% property.properties.each do |prop| -%>
+  <% property.properties.reject(&:ignore_read).each do |prop| -%>
```

Comment:
i believe we can't do this because it'll cause a diff when we set a nested object to a nested object with no value set for the ignore_read fields - we really need the value for d.Get("a.0.nested.0.field") but that's tough with hashes for sets (see https://github.com/GoogleCloudPlatform/magic-modules/issues/1019)

Reply:
Ah right. I think in another PR we should just fail on nested ignore_read since right now they appear to be silently not doing anything.

Reply:
a great idea!